src/Metrics/MetricsAggregator.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Metrics;
+
+use Sentry\Event;
+use Sentry\EventId;
+use Sentry\Metrics\Types\AbstractType;
+use Sentry\Metrics\Types\CounterType;
+use Sentry\Metrics\Types\DistributionType;
+use Sentry\Metrics\Types\GaugeType;
+use Sentry\Metrics\Types\SetType;
+use Sentry\SentrySdk;
+use Sentry\State\Scope;
+use Sentry\Tracing\TransactionSource;
+
+/**
+ * @internal
+ */
+final class MetricsAggregator
+{
+    /**
+     * @var int
+     */
+    private const ROLLUP_IN_SECONDS = 10;
+
+    /**
+     * @var array<int, AbstractType>
+     */
+    private $buckets = [];
+
+    private const METRIC_TYPES = [
+        CounterType::TYPE => CounterType::class,
+        DistributionType::TYPE => DistributionType::class,
+        GaugeType::TYPE => GaugeType::class,
+        SetType::TYPE => SetType::class,
+    ];
+
+    /**
+     * @param string[]         $tags
+     * @param int|float|string $value
+     */
+    public function add(
+        string $type,
+        string $key,
+        $value,
+        ?MetricsUnit $unit,
+        array $tags,
+        ?int $timestamp,
+        int $stackLevel,
+    ): void {
+        if ($timestamp === null) {
+            $timestamp = time();
+        }
+        if ($unit === null) {
+            $unit = MetricsUnit::none();
+        }
+
+        $tags = $this->serializeTags($tags);
+
+        $bucketTimestamp = floor($timestamp / self::ROLLUP_IN_SECONDS);
+        $bucketKey = crc32(
```

Comment:
This is risky-ish. crc32 can and will clash frequently enough. I'm not sure what the ideal option is but maybe just concatenate the fields?

Reply:
Then let's concatenate. I think md5 might become too expensive.